sfa_dash/static/js/table-search.js

```diff
@@ -1,29 +1,107 @@
-    $('.counter').text(jobCount + ' item');
+    /* Add containsi filter to JQuery selectors that selects for any word in a phrase */
+    $.extend($.expr[':'], {'containsi': function(elem, i, match, array){
+            return (elem.textContent || elem.innerText || '').toLowerCase().indexOf((match[3] || "").toLowerCase()) >= 0;
+        }
+    });
 
-  if(jobCount == '0') {$('.no-result').show();}
-    else {$('.no-result').hide();}
-		  });
+    /*
+     * Filter functions
+     *
+     * Check some state within the dom, and return a jquery object list of elements
+     * to hide.
+     */
+    function searchTable() {
+        /*
+         * Reads the value of the search input and creates a list of rows to hide
+         * based on which rows do not contain the search term
+         */
+        var searchTerm = $(".search").val();
+        var searchSplit = searchTerm.replace(/ /g, "'):containsi('")
+        return $(".results tbody tr").not(":containsi('" + searchSplit + "')")
+    }
+    function filteredOrgs() {
+        /*
+         * Iterates through the `.org-filter-option` checkboxes, for each unchecked box,
+         * builds a list of tr elements whose `.provider-column` fields contain the
+         * checkboxes value and returns the concatenated list.
+         */
+        orgsToHide = $([]);
+        $('.org-filter-option').each(function (e){
+            if (!this.checked){
+                orgsToHide = orgsToHide.add($(`.provider-column:contains("${this.value}")`).parent());
+            }
+        });
+        return orgsToHide
+    }
+    /*
+     * Filter application functions
+     */
+    var toHideFns = []
+    function allHiddenElements(){
+        /*
+         * Returns a JQuery object list of rows that need to be hidden
+         * by concatenating the results of calls to the toHideFns.
+         */
+        elements = $([]);
+        toHideFns.forEach(function(e){
+            elements = elements.add(e());
+        });
+        return elements
+    }
+    function applyTableFilters() {
+        /*
+         * Gets a list of table rows, and then builds a list of all the
+         * rows to be excluded due to filters, removes the rows to hide
+         * from all rows, sets the remainder to visible and hides the
+         * others.
+         */
+        allRows = $(".results tbody tr").not('.warning.no-result');
+        rowsToHide = allHiddenElements();
+        visibleRows = allRows.not(rowsToHide);
+        visibleRows.attr('visible', 'true');
+        rowsToHide.attr('visible', 'false');
+        
+        var jobCount = $('.results tbody tr[visible="true"]').length;
+        if(jobCount == '0') {$('.no-result').show();}
+        else {$('.no-result').hide();}
+    }
+    toHideFns.push(searchTable);
+    /*
+     * If the Provider header id exists, build a dropdown ul of checkboxes from the
+     * list of organizations in the table.
+     */
+    if ($('#provider-header').length){
```

Comment:
looks like this is mostly moved from the other file, so ok to leave as is for now. but we'll probably want to build checkbox drop down lists for at least variable, maybe others.